src/lib/mbtiles.ts

```diff
@@ -77,49 +88,120 @@ export function mbTilesToTileJSON(mbTilesDb: DatabaseInstance): TileJSON {
+): Metadata {
+  const rawMetadata: Map<string, string> = mbTilesDb
     .prepare('SELECT name, value FROM metadata')
     .all()
-    .forEach((row: { name: string; value: string }) => {
-      switch (row.name) {
-        case 'json':
-          try {
-            const jsondata = JSON.parse(row.value)
-            Object.keys(jsondata).reduce((result, key) => {
-              result[key] = result[key] || jsondata[key]
-              return result
-            }, metadata)
-          } catch (err) {
-            // TODO: Throw some error here
-          }
-          break
-        case 'minzoom':
-        case 'maxzoom':
-          metadata[row.name] = parseInt(row.value, 10)
-          break
-        case 'center':
-        case 'bounds':
-          metadata[row.name] = row.value.split(',').map(parseFloat)
-          break
-        default:
-          metadata[row.name] = row.value
-          break
+    .reduce((result, { name, value }: { name: unknown; value: unknown }) => {
+      if (typeof name === 'string' && typeof value === 'string') {
+        result.set(name, value)
+      } else {
+        console.warn('MBTiles extractor received a non-string metadata row', {
+          name,
+          value,
+        })
       }
-    })
+      return result
+    }, new Map())
+
+  const metadata: Metadata = {
+    name: rawMetadata.get('name') || fallbackName,
+
+    format: (() => {
+      const format = rawMetadata.get('format')
+      if (format && isValidMBTilesFormat(format)) return format
+      console.warn('MBTiles has an invalid (or missing) format')
+      throw new UnsupportedMBTilesFormatError()
+    })(),
+
+    bounds: parseFloatList(rawMetadata.get('bounds'), 4),
+
+    center: parseFloatList(rawMetadata.get('center'), 3),
+
+    minzoom: parseSafeInt(rawMetadata.get('minzoom')),
+
+    maxzoom: parseSafeInt(rawMetadata.get('maxzoom')),
+
+    attribution: rawMetadata.get('attribution'),
+
+    description: rawMetadata.get('description'),
 
-  // TODO: Extracted from reference implementation but not sure if it applies for us
-  // https://github.com/mapbox/node-mbtiles/blob/03220bc2fade2ba197ea2bab9cc44033f3a0b37e/lib/mbtiles.js#L300
-  metadata.scheme = 'xyz'
+    type: (() => {
+      const rawType = rawMetadata.get('type')
+      return rawType === 'overlay' || rawType === 'baselayer'
+        ? rawType
+        : undefined
+    })(),
+
+    version: rawMetadata.get('version'),
+
+    vector_layers: (() => {
+      const rawVectorLayers = parseJsonObject(
+        rawMetadata.get('json')
+      )?.vector_layers
+      return Array.isArray(rawVectorLayers) &&
+        rawVectorLayers.every((layer) => validateVectorLayerSchema(layer))
+        ? rawVectorLayers
+        : undefined
+    })(),
+
+    // TODO: Extracted from reference implementation but not sure if it applies for us
+    // https://github.com/mapbox/node-mbtiles/blob/03220bc2fade2ba197ea2bab9cc44033f3a0b37e/lib/mbtiles.js#L300
+    scheme: 'xyz',
```

Comment:
if we're not using this field anywhere else in the library, I think it's okay completely omit this. seems like it was an artifact of https://github.com/mapbox/node-mbtiles/issues/65 but I don't think it really applies to us in terms of what this function does (which is just reading the metadata).